src/library/scala/Array.scala

```diff
@@ -122,7 +122,9 @@ object Array {
     * @see `java.util.Arrays#copyOf`
     */
   def copyOf[A](original: Array[A], newLength: Int): Array[A] = (original match {
-    case x: Array[AnyRef]     => java.util.Arrays.copyOf(x, newLength)
+    case x: Array[AnyRef]     =>
+      if (x.getClass.getComponentType == classOf[BoxedUnit]) newUnitArray(newLength).asInstanceOf[Array[A]]
```

Comment:
Any reason not to use the usual match `case x: Array[BoxedUnit]`?

Reply:
I've changed to use that. The only tricky part is that it needs to precede the match on `_: Array[AnyRef]` due to array covariance in the VM.